metrics/otelhttpclient/http_transport.go

```diff
@@ -0,0 +1,134 @@
+package otelhttpclient
```

Comment:
Shall we shorten this a bit to `otelhttp`?

Reply:
otehttp is the name of the official with everything for http (server + client). i didn't prevent any confusion due to that. that's why named it as this.